src/env.cc

```diff
@@ -1153,6 +1153,17 @@ char* Environment::Reallocate(char* data, size_t old_size, size_t size) {
 
 void Environment::AddArrayBufferAllocatorToKeepAliveUntilIsolateDispose(
     std::shared_ptr<v8::ArrayBuffer::Allocator> allocator) {
+  // It is possible that we can arrive here from `node::FreeEnvironment` when it
+  // is called after `node::Stop(env)`.
+  //
+  // In this case platform's `per_isolate_` map won't have an entry for our
+  // isolate and the `AddIsolateFinishedCallback` will invoke the callback
+  // immediately leaving us with `keep_alive_allocators_` set to freed pointer.
+  if (is_stopping() && started_cleanup_) {
+    allocator.reset();
+    return;
+  }
+
```

Comment:
So … the problem is that this condition would always be true, and more or less defeat the purpose of this method. `AddArrayBufferAllocatorToKeepAliveUntilIsolateDispose()` is called from a cleanup hook, which means that `FreeEnvironment()` has been called (which is the public API that calls `Environment::RunCleanup()`), and both `is_stopping()` and `started_cleanup_` would always be set in that case.

The concerning bit here is that your comment states that this is about a missing entry in the `per_isolate_` map – neither `node::Stop(env)` nor `node::FreeEnvironment(env)` should touch that map, at all.

Apart from shutting down the platform altogether (which I assume is not happening here), entries from the map are only removed when `NodePlatform::UnregisterIsolate()` is called – so my best guess is that Electron is calling that method too early, while the `Environment` is still alive?